### PR TITLE
Make `pypeit_show_2dspec` (somewhat) backwards compatible

### PR DESCRIPTION
**Pull request by kbwestfall**
As titled.

### CODE REVIEW

File: pypeit/scripts/show_2dspec.py [hunk 1]
```diff
@@ -127,77 +131,153 @@ def main(args):
...
+                names = [h.name for h in hdu]
+                has_det = any([detname in n for n in names])
+                if not has_det:
+                    msgs.error(f'Provided file has no extensions including {detname}.')
+                for ext in ['SCIIMG', 'SKYMODEL', 'OBJMODEL', 'IVARMODEL']:
+                    _ext = f'{detname}-{ext}'
+                    if _ext not in names:
+                        msgs.error(f'{args.file} missing extension {_ext}.')
+
+                sciimg = hdu[f'{detname}-SCIIMG'].data
```
Comment (profxj) [inline pypeit/scripts/show_2dspec.py:168]: should we put this code into the spec2DObj?
  or is that simply not feasible?

File: pypeit/scripts/show_2dspec.py [hunk 2]
```diff
@@ -127,77 +131,153 @@ def main(args):
...
+                waveimg = hdu[_ext].data if _ext in names else None
+
+                img_gpm = np.ones(sciimg.shape, dtype=bool) if bpmmask is None else bpmmask == 0
+                model_gpm = img_gpm.copy()
+
+                _ext = f'{detname}-SLITS'
+                if _ext not in names:
+                    msgs.warn(f'{args.file} missing extension {_ext}; cannot show slit edges.')
+                else:
+                    slit_columns = hdu[_ext].columns.names
```
Comment (profxj) [inline pypeit/scripts/show_2dspec.py:184]: same comment as above.

and, boy, this is painful to look at ;)

**Review by profxj — APPROVED** [review]
two suggestions from me

File: pypeit/scripts/show_2dspec.py (hunk 1 above)
Comment (kbwestfall) [reply pypeit/scripts/show_2dspec.py:168]: It's feasible, but I don't think it makes sense.  I see this code as something specific to the script.  I.e., we don't want to change `Spec2DObj` (or worse `DataContainer`) more generally to allow it to distinguish between what is and is not "mission critical" for some subset of usage cases.  But this script presents a specific usage case that we're trying to accommodate.

File: pypeit/scripts/show_2dspec.py (hunk 2 above)
Comment (kbwestfall) [reply pypeit/scripts/show_2dspec.py:184]: Same response.  And, yes, agreed, although not as painful as `DataContainer` :)

Comment (jhennawi) [reply pypeit/scripts/show_2dspec.py:184]: I agree, but it is not as painful as not being able to view your reduced data.

**Review by jhennawi — APPROVED** [review]
Thanks for making these changes. I added one more spec2d obj in Slack #data_model where I encountered a failure related to flexure. Please take a look at that, but I'm approving this PR nevertheless.

**Comment by profxj** [comment]
Merge when you can, @kbwestfall

**Pull request merged**